src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -699,13 +731,47 @@ void Gather::execCompressed8Bit() {
                 size_t srcIdx = c1 + axisAndAfterAxisSize * i;
                 size_t dstIdx = c2 + specIdxAndAfterAxSize * i;
 
-                const IN_TYPE* psrc = &srcData[srcIdx];
                 OUT_TYPE* pdst = &dstData[dstIdx];
 
-                const size_t scale_offset = srcIdx / scale_group_size;
-                auto cur_zp = have_zp ? zp[srcIdx / zp_group_size] : 0;
-                for (size_t p = 0; p < afterAxisSize; p++) {
-                    pdst[p] = static_cast<OUT_TYPE>((static_cast<float>(psrc[p]) - cur_zp) * scale[scale_offset]);
+                size_t p = srcIdx;
+                size_t dst_idx = 0;
+
+                // heuristic:
+                // ((isAxisInputConst && axis == 0) && (cond1 || cond2)) take >99% probability
+                bool processed = false;
+                if (isAxisInputConst && axis == 0) {
+                    bool cond1 = have_zp && zp_group_size == scale_group_size;
+                    bool cond2 = (!have_zp) || have_scalar_zp;
+                    bool cond3 = have_scalar_scale && cond2;
+                    if (cond3) {
+                        processed = true;
+                        for (; p < srcIdx + afterAxisSize; p++) {
+                            pdst[dst_idx] = static_cast<OUT_TYPE>((static_cast<float>(srcData[p]) - zp[0]) * scale[0]);
+                            dst_idx++;
+                        }
+                    } else if (cond1 || cond2) {
```

Comment:
We don't use scalar scale for embedding compression. would propose to delete this block and let it be computed via ref path

Reply:
Yes, 
But I found this pattern with scalar scale when I fix e2e test fail issue. That model matched 3 GatherCompressed.
If you still insist, I will update it @dmitry-gorokhov

Reply:
Or as I ever said, I can add filter to disable small weight compress data for gathercompressed.
For example: weight shape[1, ?, 1024], ? should less than **1024**. because real gathercompress weight size is very big.
I can add callback and let it only works for CPU.
After adding this filter, I will remove this block, and fallback to reference. @dmitry-gorokhov

Reply:
Ok, lets leave the path, but please fix it for the case then there is no zp. Right now it always access zp[0]

Reply:
hi @dmitry-gorokhov , Please see [zp declare](https://github.com/openvinotoolkit/openvino/pull/24100/files#diff-b36e15416bd9a61ca8c78b583b3de24ae7b2f6cd8ac1893323c031b47b76276aR631). if there is no zp, zp's address will point to const_zp(0)

Reply:
oh, I see. Thanks!